src/appsettings/settings.py

```diff
@@ -837,6 +898,20 @@ def __init__(
         if max_length is not None:
             self.validators.append(MaxLengthValidator(max_length))
 
+    def decode_environ(self, value):
+        """
+        Split value by colons to a list.
+
+        Args:
+            value (str):
+
+        Returns:
+            list:
+        """
+        split_value = re.split(r"(?<!\\):", value)
+        item_convert_func = self.item_type or (lambda v: v)
+        return [item_convert_func(v.replace("\\:", ":")) for v in split_value]
```

Comment:
I'd define the separator as a class attribute.

Reply:
Done. `IterableSetting` has now `delimiter=":"` attribute, `DictSetting` has `outer_delimiter=" " inner_delimiter="="` attributes. Escaping characters has been removed since you can choose which character or string will split the original string. Splitting is done normally by `value.split(self.delimiter)`